### ISSUE TITLE
Selecting Google Maps starts Uber on Android

### ISSUE THREAD

**Issue by flyingSAP** [flexible-agency/react-native-map-link#157]
Hi,

when calling `showLocation`like so on Android using `"react-native-map-link": "^2.7.19",`

```javascript
     showLocation({
      latitude: data!.location.lat,
      longitude: data!.location.lng,
      title: 'Some TItle'
      googleForceLatLon: true,
      dialogTitle: 'Open in Maps'
      dialogMessage:  'Which app would you like to use?',
      cancelText: 'Cancel'
      appsWhiteList: undefined,
    });
```

I receive the following dialog:

![WhatsApp Image 2020-12-16 at 20 55 12](https://user-images.githubusercontent.com/26920164/102400302-05f47e00-3fe2-11eb-9387-4de6ac4423b4.jpeg)

When selecting Google Maps it opens Uber always:

![WhatsApp Image 2020-12-16 at 20 55 12 (1)](https://user-images.githubusercontent.com/26920164/102400319-0ab93200-3fe2-11eb-9404-3bc6f541eabe.jpeg)

This also happens when calling `showLocation` like so:

```javascript    
 showLocation({
      latitude: data!.location.lat,
      longitude: data!.location.lng,
      title: 'Some TItle'
      googleForceLatLon: true,
      dialogTitle: 'Open in Maps'
      dialogMessage:  'Which app would you like to use?',
      cancelText: 'Cancel'
      appsWhiteList: ['google-maps'],
    });
```

or

```javascript
     showLocation({
      latitude: data!.location.lat,
      longitude: data!.location.lng,
      title: 'Some TItle'
      googleForceLatLon: true,
      dialogTitle: 'Open in Maps'
      dialogMessage:  'Which app would you like to use?',
      cancelText: 'Cancel'
      appsWhiteList: ['google-maps'],
      app: 'google-maps'
    });
```

Any hint what I'm doing wrong?

React-Native info:

```bash
user$ npx react-native info
info Fetching system and libraries information...
System:
    OS: macOS 11.0.1
    CPU: (12) x64 Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
    Memory: 251.82 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 14.12.0 - /usr/local/bin/node
    Yarn: 1.22.5 - /usr/local/bin/yarn
    npm: 6.14.8 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.9.3 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 14.0, DriverKit 20.0, macOS 11.0, tvOS 14.0, watchOS 7.0
    Android SDK:
      API Levels: 28, 29, 30
      Build Tools: 28.0.3, 29.0.2, 29.0.3, 30.0.0
      System Images: android-28 | Intel x86 Atom_64, android-28 | Google APIs Intel x86 Atom_64, android-30 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: 4.0 AI-193.6911.18.40.6514223
    Xcode: 12.0/12A8169g - /usr/bin/xcodebuild
  Languages:
    Java: 1.8.0_261 - /usr/bin/javac
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 17.0.1 => 17.0.1 
    react-native: 0.63.3 => 0.63.3 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found
```

**Comment by Nevosis** [comment]
Same here.
To go deeper, I tried to :
`Linking.openURL(`https://maps.google.com/?q=loc:${lat},+${lng}+(${encodedTitle})`)`
And it opens uber on my coworker phone, opens googlemap on mine.

**Comment by MaxKicw** [comment]
Same here!

**Comment by tschoffelen** [comment]
Unfortunately the Uber app on Android hijacks the maps.google.com URL. Unlike on iOS, where you need to verify your app belongs to a certain domain by uploading a `apple-app-site-association` file to the server, on Android anyone can claim a certain domain belongs to their app.

So there's not much we can do about this unfortunately... Hopefully Google fixes this at some point, seems like a big security problem.

**Comment by Nevosis** [comment]
Thanks @tschoffelen 

On android, I changed thing to work with native select geoloc app by Linking to 
`geo:0,0?q=${latLng}(${title})` instead of linking to `https://maps.google.com/?q=loc.....`

**Comment by flyingSAP** [comment]
Using `geo:latitude,longitude?z=zoom` is also described in the Google Maps Intents for Android here: https://developers.google.com/maps/documentation/urls/android-intents#display_a_map

**Comment by tschoffelen** [comment]
Solved by #158
